### PR TITLE
Use Kotlin 1.9.0

### PR DESCRIPTION
**Pull request by BoD**

### CODE REVIEW

File: build-logic/src/main/kotlin/Node.kt [hunk 1]
```diff
@@ -17,10 +17,8 @@ fun Project.configureNode() {
     )
   }
 
-  // See https://youtrack.jetbrains.com/issue/KT-47215
```
Comment (BoD) [inline build-logic/src/main/kotlin/Node.kt:20]: `disableGranularWorkspaces()` no longer exists, but it looks like the original issue KT-47215 is also no longer relevant

File: tests/defer/src/jsTest/kotlin/util/HttpUtil.kt [hunk 2]
```diff
@@ -1,6 +0,0 @@
-package util
-
-import http.OutgoingHttpHeaders
```
Comment (BoD) [inline tests/defer/src/jsTest/kotlin/util/HttpUtil.kt:3]: This import was invalid it seems, but this whole class is unused anyway

File: tests/defer/src/jsTest/kotlin/util/JsUtil.kt [hunk 3]
```diff
@@ -1,8 +1,11 @@
 package util
 
-import tsstdlib.Symbol
```
Comment (BoD) [inline tests/defer/src/jsTest/kotlin/util/JsUtil.kt:3]: This import was invalid, not sure why it didn't cause a problem before

File: libraries/apollo-gradle-plugin/testProjects/multiplatform/build.gradle.kts [hunk 4]
```diff
@@ -26,18 +26,3 @@ apollo {
         packageNamesFromFilePaths()
     }
 }
-
-// See https://youtrack.jetbrains.com/issue/KT-56019
-val myAttribute = Attribute.of("com.apollographql.test", String::class.java)
```
Comment (martinbonnin) [inline libraries/apollo-gradle-plugin/testProjects/multiplatform/build.gradle.kts:31]: I think we have more instances of this in `tests`

File: libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/checkConditionalFragment.kt [hunk 5]
```diff
@@ -21,6 +21,7 @@ internal fun checkConditionalFragments(definitions: List<GQLDefinition>): List<I
     when (it) {
       is GQLOperationDefinition -> scope.checkConditionalFragments(it.selectionSet.selections)
       is GQLFragmentDefinition -> scope.checkConditionalFragments(it.selectionSet.selections)
+      else -> {}
```
Comment (martinbonnin) [inline libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/checkConditionalFragment.kt:24]: Interesting. Is that triggering an error now or is it a warning we've had forever?

File: tests/defer/src/jsTest/kotlin/test/DeferWithHelixTest.kt [hunk 6]
```diff
@@ -1,285 +0,0 @@
...
-import kotlin.random.Random
-import kotlin.test.Ignore
-import kotlin.test.Test
-import kotlin.test.assertEquals
-
-@Ignore
-// TODO These tests are temporarily ignored as the latest version of graphql-js
-// uses a package format which is incompatible with Kotlin/JS.
-// See https://youtrack.jetbrains.com/issue/KT-12784
-class DeferWithHelixTest {
```
[the remark this answers is not in the document]
Comment (BoD) [reply tests/defer/src/jsTest/kotlin/test/DeferWithHelixTest.kt:38]: There were errors related to JS dependencies in this area and it took me a while to figure out it was related to `disableGranularWorkspaces` (see above). My first attemt was to get rid of the issue altogether (as these tests have been ignored for a while anyway). Now looking at KT-12784 it is possible that we could re-enable them actually 🤔 I'll undelete them and have a look at if we can unignore them!

File: libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/checkConditionalFragment.kt (hunk 5 above)
Comment (BoD) [reply libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/checkConditionalFragment.kt:24]: Not sure why but now if I remove this now it compiles fine - I'm pretty sure it didn't while I was working on this PR 🤨 There is a warning in the IDE only though.

Comment (martinbonnin) [reply libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/checkConditionalFragment.kt:24]: The plot thickens... 🕵️ Not a big deal, was just curious. Keeping the `else` branch is better 👍

File: libraries/apollo-gradle-plugin/testProjects/multiplatform/build.gradle.kts (hunk 4 above)
Comment (BoD) [reply libraries/apollo-gradle-plugin/testProjects/multiplatform/build.gradle.kts:31]: Actually removing the ones in tests still fails. But this one produces `Configuration with name 'releaseFrameworkIosFat' not found.` and removing this resolves that.

Comment (martinbonnin) [reply libraries/apollo-gradle-plugin/testProjects/multiplatform/build.gradle.kts:31]: Interesting, looks like [it's not yet fixed with 1.9](https://youtrack.jetbrains.com/issue/KT-55751/MPP-Gradle-Consumable-configurations-must-have-unique-attributes#focus=Comments-27-7362740.0-0). Something else to try in the -RC

Comment (martinbonnin) [reply libraries/apollo-gradle-plugin/testProjects/multiplatform/build.gradle.kts:31]: Added a comment there: https://youtrack.jetbrains.com/issue/KT-55751/MPP-Gradle-Consumable-configurations-must-have-unique-attributes#focus=Comments-27-7417705.0-0

File: tests/defer/src/jsTest/kotlin/test/DeferWithHelixTest.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (BoD) [reply tests/defer/src/jsTest/kotlin/test/DeferWithHelixTest.kt]: I undeleted them, but AFAICT using the latest version of GraphqL-JS (ESM) still doesn't work - added a comment on KT-12784.

File: build-logic/src/main/kotlin/Node.kt (hunk 1 above)
Comment (BoD) [reply build-logic/src/main/kotlin/Node.kt:20]: Using a dev version of the RC fixes it.

**Review by martinbonnin — COMMENTED** [review]
I was going to comment that we should merge this but I'm having second thoughts on the K/N binary format. I wouldn't want to force all K/N users on 1.9-RC so my vote goes to waiting for stable before merging.

**Comment by BoD** [comment]
>I was going to comment that we should merge this but I'm having second thoughts on the K/N binary format. I wouldn't want to force all K/N users on 1.9-RC so my vote goes to waiting for stable before merging.

Ah yeah it's true about the native binaries.

Yes, makes sense to wait, and it shouldn't be too long before the stable anyway.

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**